lxd/instance/drivers/driver_lxc.go

```diff
@@ -2048,23 +2070,31 @@ func (d *lxc) startCommon() (string, []func() error, error) {
 				}
 			}
 
-			if d.state.OS.Shiftfs && !d.IsPrivileged() && diskIdmap == nil {
-				// Host side mark mount.
-				err = lxcSetConfigItem(d.c, "lxc.hook.pre-start", fmt.Sprintf("/bin/mount -t shiftfs -o mark,passthrough=3 %s %s", strconv.Quote(d.RootfsPath()), strconv.Quote(d.RootfsPath())))
-				if err != nil {
-					return "", nil, errors.Wrapf(err, "Failed to setup device mount shiftfs '%s'", dev.Name)
-				}
+			if !d.IsPrivileged() && diskIdmap == nil {
+				idmapType := d.IdmappedStorage()
```

Comment:
@brauner can you reuse the idmapType var from line 1908 here instead of an additional call to `IdMappedStorage()`? https://github.com/lxc/lxd/pull/8778/commits/f02ed955349d2f08ae5f5c32857518ae75f1abee#diff-5d42fb3e7face87a44e52b8071be7823459e2360ba2c0349029a5e4b4dba9426R1908